contents/blog/how-we-built-web-analytics.md

```diff
@@ -0,0 +1,118 @@
+
+Alas, this is not the case. It took <TeamMember name="Robbie Coomber" /> a year to do it, and it's not because he's lazy. 
+
+Fundametally, they have different use cases and thus require different architectures. In this post, we dive into the differences between the two, the challenges we faced, and how we solved them.
+
+![Photo of Robbie being lazy](placeholder)
+<Caption>Robbie. He's definitely not lazy.</Caption>
+
+## The difference between web and product analytics
+
+When people talk about **web analytics**, they're referring to tracking visitors on your marketing website. They care about things like page views, bounce rate, and top sources of traffic.
+
+On the other hand, **product analytics** tracks how people are using your web or mobile app. It gives you insights into metrics such as feature adoption rate, user retention, and churn.
+
+In practice, this means that for web analytics you care about _what_ people are doing on your website and _how many_ people are doing it, whereas in product analytics the _who_ is also important.
+
+![Web analytics vs product analytics](https://res.cloudinary.com/dmukukwp6/image/upload/web_vs_prodyuct_21a8b644e9.png)
+
+Fundamentally, this makes web analytics simpler, and simpler means cheaper and faster. So if we used our existing product analytics architecture for web analytics, users would complain about the performance and costs.
+
+To understand the reasons why, let's dive into each one in detail.
+
+### 1. Why web analytics is cheaper
+
+In web analytics, everything is session-based. A session consists of multiple events. Each event has properties set on it, such as the URL, device, and geolocation. When you query data, properties are read from the event. 
+
+This also means that in web analytics, there's no concept of a "person". So you cannot query based on person properties. For example, you cannot filter queries based on the date a user first signed up.
+
+Contrast this to product analytics, where each user has a person profile. This means that not only do events have properties, but so do people. For example, you might want to query based on the date a user first signed up. 
+
+![Web events vs product events](https://res.cloudinary.com/dmukukwp6/image/upload/events_6caabf2705.png)
+
+Person properties change frequently too, since every event may have data that could be relevant to the user. For example, the URL of the last page they visited, or the most recent browser they used are important properties to store.
+
+This requires creating and storing a person profile for each user. Usually, we use [ClickHouse](https://posthog.com/blog/how-we-turned-clickhouse-into-our-eventmansion) for our data storage. However, ClickHouse is optimized for storing larges amounts of data that never changes (like events). But since person properties can change frequently, we store them in Postgres.
```

Comment:
AFAIK "Person on events" means we store person properties in ClickHouse as well.